src/sentry/api/utils.py

```diff
@@ -486,5 +490,13 @@ def update_snuba_params_with_timestamp(
         example_end = example_timestamp + timedelta(days=time_buffer)
         # If timestamp is being passed it should always overwrite the statsperiod or start & end
         # the client should just not pass a timestamp if we need to overwrite this logic for any reason
-        params["start"] = max(params["start"], example_start)
-        params["end"] = min(params["end"], example_end)
+        if isinstance(params, SnubaParams):
+            # Typing gets mad that start is optional still but has_dates has checked it
+            params.start = cast(datetime.datetime, params.start)
+            params.end = cast(datetime.datetime, params.end)
```

Comment:
Not a huge fan of casting if we avoid it as it makes assumptions in the preceding code. Can this be an assertion instead?

```python
assert params.start is not None
assert params.end is not None
```

Reply:
For sure, good idea